src/input/input_python.rs

```diff
@@ -403,6 +432,14 @@ impl<'py> Input<'py> for Bound<'py, PyAny> {
     }
 
     fn lax_dict<'a>(&'a self) -> ValResult<GenericPyMapping<'a, 'py>> {
+
```

Comment:
I think you will need to fix `strict_dict` too.

Probably rather than checking specifically for `OrderedDict` you should return `GenericPyMapping::Mapping` for all cases where it's a subclass of dict.

Reply:
alright I did that. For the `strict_dict` this approach doesn't work since we explicitly check explicitly that an error is thrown if we hand over a `Mapping` (see [here](https://github.com/pydantic/pydantic-core/blob/main/tests/validators/test_dict.py#L123)), and it makes sense in strict mode to only allow dicts. To not break code (OrderedDicts were previously working, just didn't keep order), I added an explict check for `OrderedDict` in the fashion I previously used for `lax_dict`. Let me know what you think here